eldritch/assets.py

```diff
@@ -273,6 +276,57 @@ def get_trigger(self, event, owner, state):
     return None
 
 
+class SelfDiscardingCard(Card):
+  def __init__(self, name, idx, active_bonuses=None, passive_bonuses=None):
+    active_bonuses = active_bonuses or {}
+    passive_bonuses = passive_bonuses or {}
+    super().__init__(
+        name, idx, "specials", active_bonuses=active_bonuses, passive_bonuses=passive_bonuses
+    )
+    self.tokens["elder_sign"] = 0
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.UpkeepActions) and event.character == owner:
+      if self.tokens["elder_sign"]:
+        return events.RollToLose(owner, self)
+      return events.AddToken(self, "elder_sign", owner)
+    return None
+
+
+class BlessingOrCurse(SelfDiscardingCard):
+  def __init__(self, name, idx):
+    assert name in ["Blessing", "Curse"]
+    super().__init__(name, idx)
+    self.opposite = "Curse" if name == "Blessing" else "Blessing"
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.KeepDrawn) and self.name in event.kept:
+      selves = [p for p in event.character.possessions if p.name == self.name]
+      kept, *duplicates = sorted(selves, key=lambda x: x.tokens["elder_sign"])
+      if duplicates:
+        return events.Sequence([
+            events.DiscardSpecific(event.character, duplicates),
+            events.RemoveToken(kept, "elder_sign", owner),
+        ], owner)
+      if self.opposite in [p.name for p in event.character.possessions]:
```

Comment:
How will this interact with the nun's personal story? If she gains a curse card, she fails. But in this case, she would gain a curse card, and then immediately discard it.

The more I look at the different possibilities here, the more I think we may actually need to hard-code this interaction (into KeepDrawn, maybe?). I can't think of an elegant way to do it.